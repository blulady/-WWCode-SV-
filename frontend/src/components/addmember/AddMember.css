/* Start of Chapter Member Button */
.button-inside-div {
  background: rgba(130, 130, 130, 0.1);
  border-radius: 8px;
  display: flex;
  align-items: center;
  margin: 20px 0px 20px 0px;
}
.chapter-member-btn {
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  text-align: center;
  color: #0f283c;
}
/* End of Chapter Member Button */

/* Start of Form div */
.form-div {
  background: #ffffff !important;
  border-radius: 10px;
  padding: 20px 20px 20px 20px;
  /* overflow: hidden; */
}
/* End of Form div */

/* Start of back to Member Portal Button */
.back-member-img-btn-div {
  background: #ffffff;
  border: 1px solid #007a7c;
  box-sizing: border-box;
  border-radius: 5px;
  display: flex;
  flex-direction: row;
  align-items: center;
  padding-left: 10px;
}

.back-member-btn {
  font-family: Open Sans;
  font-style: normal;
  text-align: center;
  color: #007a7c;
  display: flex;
  align-items: center;
}
/* End of back to Member Portal Button */

.header {
  font-family: Montserrat;
  font-style: normal;
  font-weight: bold;
  text-align: center;
  color: #0a0505;
}
.subheader {
  margin-top: 5px;
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  text-align: center;
  color: #48484a;
}

.email-input {
  background: #ffffff;
  border: 1px solid #333333;
  box-sizing: border-box;
}

.input-label {
  font-family: Open Sans;
  font-style: normal;
  font-weight: 600;
  font-size: 14px;
  line-height: 18px;
  color: #333333;
  padding-left: 0px !important;
}

.role-radio {
  width: 15px;
  height: 15px;
  background: #ffffff;
  border: 1px solid #333333;
  box-sizing: border-box;
  margin-left: 5px;
}

.member-role {
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 19px;
  color: #48484a;
}

.role-desc {
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  color: #007d7f;
}
.radio-input {
  margin-bottom: 10px !important;
  background: #f5f5f5;
  border-radius: 5px;
}
.radio-btn-selected {
  margin-bottom: 10px !important;
  background: #dbebeb;
  border: 2px solid #007a7c;
  box-sizing: border-box;
  border-radius: 5px;
}
.role-radio-selected {
  width: 15px;
  height: 15px;
  margin-left: 5px;
  background: #007d7f;
}
.message-textarea {
  background: #ffffff;
  border: 1px solid #333333;
  box-sizing: border-box;
}
.review-btn {
  width: 155px;
  height: 39px;
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  display: flex;
  align-items: center;
  text-align: center;
  background: #9f9f9f;
  border-radius: 5px;
  color: #ffffff;
}
.review-btn:enabled {
  width: 140px;
  height: 39px;
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  display: flex;
  align-items: center;
  text-align: center;
  background: #007a7c;
  border-radius: 5px;
  color: #ffffff;
}

.message-counter {
  float: right;
  align-items: end;
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  text-align: right;
  color: #333333;
}
.confirm-message {
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  text-align: center;
  color: #007d7f;
}

.modal-open-text {
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  text-align: center;
  background: #ffffff;
  box-shadow: 0px 10px 24px 2px rgba(0, 0, 0, 0.1);
  border-radius: 5px;
  color: #000000;
}
.modal-confirm-btn {
  margin: 20px;
  width: 129px;
  height: 39px;
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  text-align: center;
  color: #ffffff;
  background: #007a7c;
  border-radius: 5px;
}
.modal-success-header {
  border-bottom: 0px !important;
  padding-bottom: 0px !important;
}
.modal-success-title {
  font-family: Montserrat;
  font-style: normal;
  font-weight: bold;
  font-size: 30px;
  line-height: 37px;
  text-align: center;
  align-items: center !important;
  border-bottom: 0px;
  color: #007a7c;
  padding-bottom: 1.5rem;
}

.modal-success-body {
  padding: 0 !important;
}
.modal-success-bg {
  background-color: rgba(0, 0, 0, 0.4) !important;
}
/*End of Form div */

/* Smaller Resolution */
@media only screen and (max-width: 900px) {
  .button-inside-div {
    width: 100px;
    height: 20px;
  }
  .chapter-member-btn {
    font-size: 10px;
    line-height: 20px;
    width: 100px;
    height: 20px;
  }

  .back-member-img-btn-div {
    width: 160px;
    height: 30px;
  }
  .back-btn-img {
    width: 10.39px !important;
    height: 10.12px !important;
  }
  .back-member-btn {
    font-weight: normal;
    font-size: 12px;
    line-height: 16px;
    padding: 1px 1px 1px 1px;
  }
  .form-div {
    width: 306px;
    height: 90vh;
    margin: auto;
    overflow: hidden;
  }
  .form-div-spacing {
    margin-top: 10px;
  }
  .header {
    font-size: 16px;
    line-height: 16px;
  }
  .subheader {
    font-size: 10px;
    line-height: 14px;
  }
  .member-role {
    width: 84px;
    height: 19px;
    font-family: Open Sans;
    font-style: normal;
    font-weight: 600;
    font-size: 14px;
    line-height: 19px;
    margin-left: 8px;
    color: #48484a;
  }
  .role-desc {
    width: 212px;
    height: 23px;
    font-size: 10px;
    line-height: 14px;
    margin-left: 24px;
  }
  .role-radio {
    transform: translate(0%, 60%);
    margin-left: 0px !important;
  }
  .role-radio-selected {
    transform: translate(0%, 60%);
    margin-left: 0px !important;
  }
  .radio-input {
    width: 261px;
    height: 35px;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
  }
  .radio-btn-selected {
    width: 261px;
    height: 38px;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
  }
  .email-input {
    width: 261.39px;
    height: 30px;
  }
  .message-textarea {
    width: 261.39px !important;
    height: 115px !important;
  }
  #confirmModal {
    outline: none;
    overflow: hidden;
  }
  .modal-success-btn {
    margin: 20px;
    width: 51px;
    height: 29.39px;

    background: #ffffff;
    border: 1px solid #007a7c;
    box-sizing: border-box;
    border-radius: 5px;

    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 16px;
    align-items: center;
    text-align: center;
    color: #007a7c;
  }
  .modal-success-body-text {
    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 13px;
    line-height: 18px;
    text-align: center;
  }
  .modal-success-box {
    width: 306px;
    height: 168px;
  }
  .review-email-div {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    padding-left: 15px;
    padding-right: 15px;
    width: 280px;
    height: 110px;
    background: #f5f5f5;
  }
  .email-editicon-transform {
    transform: translate(200px, -60px);
  }
  .role-editicon-transform {
    transform: translate(200px, -70px);
  }
  .message-editicon-transform {
    transform: translate(200px, -65px);
  }
  .review-role-div {
    width: 280px;
    min-height: 110px;
    background: #ffffff;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    padding-left: 15px;
    padding-right: 15px;
  }

  .responsive-div-block {
    display: block !important;
  }

  .review-role-desc {
    margin-left: 0px !important;
  }
  .review-message-div {
    width: 280px;
    min-height: 100px;
    background: #f5f5f5;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    padding-left: 15px;
    padding-right: 15px;
  }
  .responsive-save-btn {
    float: right;
  }
}
/* Bigger Resolution */
@media only screen and (min-width: 905px) {
  .button-inside-div {
    width: 155px;
    height: 51px;
  }
  .chapter-member-btn {
    width: 150px;
    height: 22.44px;
    font-size: 16px;
    line-height: 22px;
  }
  .back-member-img-btn-div {
    width: 229px;
    height: 34px;
  }
  .back-member-btn {
    font-weight: 600;
    font-size: 16px;
    line-height: 22px;
  }
  .form-div {
    width: 1080px; 
    height: 80vh; 
    margin: 20px 0px 20px 0px;
  }

  .header {
    font-size: 25px;
    line-height: 25px;
  }
  .subheader {
    font-size: 14px;
    line-height: 19px;
  }
  .role-desc {
    font-size: 11px;
    line-height: 15px;
    display: flex;
    align-items: center;
    margin-left: 20px;
  }
  .radio-input {
    padding: 5px;
    width: 408.17px;
    height: 30px;
    display: flex;
    align-items: center;
  }
  .radio-btn-selected {
    width: 408.17px;
    height: 30px;
    display: flex;
    align-items: center;
    color: #007a7c;
  }
  .email-input {
    width: 408.17px;
  }
  .message-textarea {
    width: 408.17px !important;
    height: 100px !important;
  }
  .member-role {
    width: 64px;
    height: 19px;
    margin-left: 10px;
  }
  #confirmModal {
    /* top: 60%;
    right: 50%; */
    outline: none;
    overflow: hidden;
  }
  .modal-success-btn {
    margin: 20px;
    width: 108px;
    height: 39px;

    background: #ffffff;
    border: 1px solid #007a7c;
    box-sizing: border-box;
    border-radius: 5px;

    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 16px;
    line-height: 22px;
    align-items: center;
    text-align: center;
    color: #007a7c;
  }
  .modal-success-body-text {
    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 16px;
    line-height: 22px;
    text-align: center;
  }
  .review-email-div {
    width: 730px;
    height: 85px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    padding-left: 75px;
    padding-right: 75px;
    background: #f5f5f5;
  }
  .review-role-desc {
    margin-left: 0px !important;
  }

  .review-role-div {
    width: 730px;
    min-height: 85px;
    background: #ffffff;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    padding-left: 75px;
    padding-right: 75px;
  }
  .review-message-div {
    width: 730px;
    min-height: 130px;
    background: #f5f5f5;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    padding-left: 75px;
    padding-right: 75px;
  }
  .save-btn {
    width: 63px;
    height: 35px;
    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 15px;
    line-height: 20px;
    display: flex;
    align-items: center;
    color: #007a7c;
  }
}

/* Additional classes for Review Form CSS */
.padded {
  padding: 1em;
}
.centered {
  display: flex;
  justify-content: center;
  align-items: center;
}
.edit-icon {
  width: 18px;
  height: 18px;
}
.save-btn {
  background: #ffffff;
  border: 1px solid #007a7c;
  box-sizing: border-box;
  border-radius: 5px;
}
.spacer {
  flex: 1;
}
.hide {
  display: none;
}
.show {
  display: block;
}
.reviewform-border {
  border: 2px solid #fafbfc;
  box-sizing: border-box;
}

.div-inline {
  display: inline-flex;
}

.sendinvite-btn {
  width: 155px;
  height: 39px;
  margin: 15px;
  background: #007a7c;
  border-radius: 5px;
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  align-items: center;
  text-align: center;
  color: #ffffff;
}
.review-member-role {
  width: 64px;
  height: 19px;
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 19px;
  color: #48484a;
}
.review-role-desc {
  width: 304px;
  height: 23px;
  font-family: Open Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 11px;
  line-height: 15px;
  margin-left: 20px;
  color: #007d7f;
}
