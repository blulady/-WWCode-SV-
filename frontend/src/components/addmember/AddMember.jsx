import React, { useState } from "react";
import "./AddMember.css";
import { useNavigate, useLocation } from "react-router-dom";
import ContainerWithNav from "../layout/ContainerWithNav";
import BackButton from "../../images/arrow_back_24px.png";
import EmailInput from "./EmailInput";
import RoleRadioInput from "./RoleRadioField";
import InputLabel from "./InputLabel";
import TextAreaInput from "./TextAreaInput";
import ConfirmationModal from "./ConfirmationModal";
import SuccessModal from "./SuccessModal";
import BackToMemberPortal from "../layout/BackToMemberPortal";

function AddMember(props) {
  const navigate = useNavigate();
  const location = useLocation();
  const teamId = (location.state ? location.state.teamId : 0);
  const pending = (location.state ? location.state.pending : false);
  let successFlag = false;
  if (location.state) {
    successFlag = location.state.fromReview;
  }

  const [memberRole, setMemberRole] = useState({
    MemberRoleId: "",
    MemberRole: "",
    MemberRoleDescription: "",
  });

  const [newMember, setNewMember] = useState({
    Email: "",
    Role: "",
    Message: "",
  });

  const MemberRoleArray = [
    { id: "1", role: "Volunteer", desc: "Limited access to areas of portal" },
    {
      id: "2",
      role: "Leader",
      desc: "Access to all areas excluding Director area",
    },
    { id: "3", role: "Director", desc: "Access to all areas of portal" },
  ];
  const [check, setCheck] = useState("");
  const [showModal, setShowModal] = useState(false);
  const [showSuccessModal, setShowSuccessModal] = useState(
    successFlag ? true : false
  );
  const handleRadioClick = (id) => {
    MemberRoleArray.forEach((item) => {
      if (item.id === id) {
        setMemberRole({
          MemberRoleId: item.id,
          MemberRole: item.role,
          MemberRoleDescription: item.desc,
        });
        setCheck(id);
        setShowModal(true);
      }
    });
  };

  const handleChange = (event) => {
    // event.persist();
    setNewMember({ ...newMember, [event.target.name]: event.target.value });
  };

  const handleSubmit = (event) => {
    navigate( "/member/review", {
      state: { memberinfo: newMember, roleinfo: memberRole, teamId, pending },
    });
  };

  return (
    <ContainerWithNav>
      <main>
        <div className='container'>
          <div className='button-inside-div'>
            <button
              className='chapter-member-btn'
              onClick={() => {
                navigate("/team/0/members");
              }}
            >
              {" "}
              Chapter Members{" "}
            </button>
          </div>
          <div className='form-div'>
            <BackToMemberPortal teamId={teamId} pending={pending}/>
            <div className='row justify-content-center form-div-spacing'>
              <div>
                <div className='header'>Add New Member</div>
                <div className='header'>to Portal</div>
                <p className='subheader'> *Mandatory Fields</p>
                <form onSubmit={handleSubmit}>
                  <div className='form-group'>
                    <InputLabel labelName='email' text='Email *' />
                    <EmailInput
                      name='Email'
                      className='form-control email-input'
                      pclass='hide'
                      editclass='hide'
                      buttonclass='hide'
                      onChange={handleChange}
                      placeholder='eg. sam@wwcode.com'
                    />
                  </div>
                  <div className='form-group'>
                    <InputLabel labelName='role' text='Role *' />
                    {MemberRoleArray.map((item) => (
                      <div
                        key={item.id}
                        className={
                          check === item.id
                            ? "form-check radio-btn-selected"
                            : "form-check radio-input"
                        }
                      >
                        <RoleRadioInput
                          className={
                            check === item.id
                              ? "form-check-input position-static role-radio-selected"
                              : "form-check-input position-static role-radio"
                          }
                          id={item.id}
                          name='Role'
                          value={item.id}
                          pclass='hide'
                          datatarget='#confirmModal'
                          onClick={() => handleRadioClick(item.id)}
                          onChange={handleChange}
                          roletext={item.role}
                          roledesc={item.desc}
                        />
                      </div>
                    ))}
                  </div>
                  <div className='form-group'>
                    <InputLabel labelName='message' text='Message (optional)' />
                    <TextAreaInput
                      name='Message'
                      pclass='hide'
                      editclass='hide'
                      buttonclass='hide'
                      className='form-control message-textarea'
                      onChange={handleChange}
                      counterclass='message-counter'
                      value={newMember.Message}
                      countervalue={
                        newMember.Message
                          ? newMember.Message.length + "/2000 char"
                          : null
                      }
                    />
                  </div>
                  <div className='form-group d-flex justify-content-center'>
                    <button
                      type='submit'
                      disabled={!newMember.Email || !newMember.Role}
                      className='d-flex justify-content-center review-btn'
                    >
                      Review
                    </button>
                  </div>
                </form>
                {/* Role Confirmation Modal code starts */}
                <ConfirmationModal
                  memberrole={memberRole.MemberRole}
                  memberdesc={memberRole.MemberRoleDescription}
                />
                {/* Role Confirmation Modal code ends */}
                {/* Success Modal code starts */}
                {showSuccessModal ? (
                  <SuccessModal
                    onClick={() => {
                      setShowSuccessModal(false);
                      navigate("", { replace: true })
                    }}
                  />
                ) : null}
                {/* Success Modal code ends */}
              </div>
            </div>
          </div>
        </div>
      </main>
    </ContainerWithNav>
  );
}
export default AddMember;
