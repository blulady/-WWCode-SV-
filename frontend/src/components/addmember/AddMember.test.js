import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import AddMember from './AddMember';

const mockSetState = jest.fn();
const mockHistoryPush = jest.fn();
const expectedMemberInfo = {
    Email: 'test@example.com',
    Role: 'Leader',
    Message: 'hello world!'
};
const expectedMemberRole = {
    'MemberRoleId': '2',
    'MemberRole': 'Leader',
    'MemberRoleDescription': 'Access to all areas excluding Director area'
};

jest.mock('react-router-dom', () => ({
    useHistory: () => ({
      push: mockHistoryPush,
      replace: jest.fn()
    }),
    useLocation: () => ({
        state: {
            fromReview: false
        }
    })
}));

jest.mock('react', () => {
    const ActualReact = require.requireActual('react');
    return {
        ...ActualReact,
        useContext: () => ({
            userInfo: {
                email: 'director@example.com',
                first_name: 'John',
                id: 1,
                last_name: 'Smith',
                role: 'DIRECTOR'
            },
            handleRemoveAuth: jest.fn()
        }),
        useState: (init) => {
            return [init, mockSetState];
        }
    };
});

describe('AddMember', () => {
    beforeEach(() => {
        mockSetState.mockClear();
    });

    test('it renders without crashing', () => {
        const { container } = render(<AddMember />);
        
        expect(container).toMatchSnapshot();
    });

    test('it calls setNewMember on input change', () => {
        const { container } = render(<AddMember />);
        const emailInput = container.querySelector('.email-input');
        const selectedRole = container.querySelector("[id='2']");
        const textAreaInput = container.querySelector('.message-textarea');

        act(() => {
            fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
            fireEvent.click(selectedRole);
            fireEvent.change(textAreaInput, { target: { value: 'hello world!' } });
        });

        expect(mockSetState).toHaveBeenCalledTimes(6);
        // expect to have been called with
    });

    test('it calls setMemberRole, setNewMember, setCheckId, setShowModal on radio click', () => {
        const { container } = render(<AddMember />);
        const selectedRole = container.querySelector("[id='2']");

        act(() => {
            fireEvent.click(selectedRole);
        });

        expect(mockSetState).toHaveBeenCalledTimes(4);

        // setMemberRole call
        //expect(mockSetState).toBeCalledWith(expectedMemberRole);

        // setCheckId call
        //expect(mockSetState).toBeCalledWith(2);

        // setShowModal call
        //expect(mockSetState).toBeCalledWith(true);

        // setNewMember call
        //expect(mockSetState).toBeCalledWith({ Role: 'Leader' });
    });

    test('it calls handleSubmit on form submit', () => {
        const { container } = render(<AddMember />);
        const emailInput = container.querySelector('.email-input');
        const selectedRole = container.querySelector("[id='3']");
        const reviewBtn = container.querySelector('.review-btn');

        act(() => {
            fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
            fireEvent.click(selectedRole);
        });

        act(() => {
            const confirmBtn = container.querySelector('.modal-confirm-btn');
            fireEvent.click(confirmBtn);
            fireEvent.click(reviewBtn);
        });

        //expect(mockHistoryPush).toHaveBeenCalledTimes(1);
        //expect(mockHistoryPush).toBeCalledWith({ pathname: '/member/review', state: { memberinfo: expectedMemberInfo, roleinfo: expectedMemberRole } });
    });
});