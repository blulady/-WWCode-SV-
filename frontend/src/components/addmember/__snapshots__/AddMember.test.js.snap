// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMember it renders without crashing 1`] = `
<div>
  <div
    class="container-fluid container-fluid"
  >
    <nav
      class="container-navbar"
    >
      <div
        class="container-inner"
      >
        <div
          class="navbar-brand navbar-wwc-logo"
        >
          <div
            class="logo-png"
          />
        </div>
        <div
          class="navbar-items "
        >
          <ul
            class="navbar-items-ul"
          >
            <li
              class="navbar-text navbar-items-wwc-title"
            >
              Chapter Tools
            </li>
            <li
              class="navbar-items-user-button"
            >
              <div
                aria-expanded="false"
                aria-haspopup="true"
                class="user-button"
                data-bs-toggle="dropdown"
                id="navbarDropdown"
                role="button"
              >
                <span>
                  J
                </span>
              </div>
              <div
                aria-labelledby="navbarDropdown"
                class="dropdown-menu dropdown-menu"
              >
                <p
                  class="dropdown-item signed-user dropdown-item"
                >
                  Signed-in as
                </p>
                <p
                  class="font-weight-bold dropdown-item signed-user dropdown-item"
                >
                  director@example.com
                </p>
                <button
                  class="dropdown-item dropdown-item"
                >
                  Your Profile
                </button>
                <hr
                  class="dropdown-divider dropdown-hr"
                />
                <button
                  class="dropdown-item item-logout dropdown-item"
                >
                  Log Out
                </button>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      class="container-body"
    >
      <main>
        <div
          class="container"
        >
          <div
            class="button-inside-div"
          >
            <button
              class="chapter-member-btn"
            >
               
              Chapter Members
               
            </button>
          </div>
          <div
            class="form-div"
          >
            <div>
              <button
                class="back-member-btn"
              >
                <img
                  alt="Back Button"
                  class="back-btn-img"
                  src="arrow_back_24px.png"
                />
                Back to 
                Chapter Members
              </button>
            </div>
            <div
              class="row justify-content-center form-div-spacing"
            >
              <div
                class="form-inner"
              >
                <div
                  class="header"
                >
                  Add New Member
                </div>
                <div
                  class="header"
                >
                  to Portal
                </div>
                <p
                  class="subheader"
                >
                   *Mandatory Fields
                </p>
                <form>
                  <div
                    class="form-group"
                  >
                    <div>
                      <label
                        class="input-label"
                      >
                        Email *
                      </label>
                    </div>
                    <div
                      class="div-inline responsive-div-block"
                    >
                      <p
                        class="hide"
                      />
                      <input
                        class="form-control email-input"
                        id="email"
                        name="Email"
                        placeholder="eg. sam@wwcode.com"
                        required=""
                        type="email"
                        value=""
                      />
                      <div
                        class="spacer"
                      />
                      <div
                        class="edit-icon"
                      >
                        <img
                          alt="Edit"
                          class="hide"
                          src="edit_24px.png"
                          style="width: auto; height: auto;"
                        />
                      </div>
                      <button
                        class="hide"
                        type="submit"
                      >
                        Save
                      </button>
                    </div>
                  </div>
                  <div
                    class="form-group"
                  >
                    <div>
                      <label
                        class="input-label"
                      >
                        Role *
                      </label>
                    </div>
                    <div
                      class="form-check radio-input"
                    >
                      <input
                        class="form-check-input position-static role-radio"
                        data-bs-target="#confirmModal"
                        data-bs-toggle="modal"
                        id="1"
                        name="Role"
                        type="radio"
                        value="1"
                      />
                      <div
                        class="member-role"
                      >
                        Volunteer
                      </div>
                      <div
                        class="role-desc"
                      >
                        Limited access to areas of portal
                      </div>
                    </div>
                    <div
                      class="form-check radio-input"
                    >
                      <input
                        class="form-check-input position-static role-radio"
                        data-bs-target="#confirmModal"
                        data-bs-toggle="modal"
                        id="2"
                        name="Role"
                        type="radio"
                        value="2"
                      />
                      <div
                        class="member-role"
                      >
                        Leader
                      </div>
                      <div
                        class="role-desc"
                      >
                        Access to all areas excluding Director area
                      </div>
                    </div>
                    <div
                      class="form-check radio-input"
                    >
                      <input
                        class="form-check-input position-static role-radio"
                        data-bs-target="#confirmModal"
                        data-bs-toggle="modal"
                        id="3"
                        name="Role"
                        type="radio"
                        value="3"
                      />
                      <div
                        class="member-role"
                      >
                        Director
                      </div>
                      <div
                        class="role-desc"
                      >
                        Access to all areas of portal
                      </div>
                    </div>
                  </div>
                  <div
                    class="form-group"
                  >
                    <div>
                      <label
                        class="input-label"
                      >
                        Message (optional)
                      </label>
                    </div>
                    <div
                      class="div-inline responsive-div-block"
                    >
                      <p />
                      <div
                        style="display: block;"
                      >
                        <textarea
                          class="form-control message-textarea"
                          data-testid="message"
                          id="message"
                          maxlength="2000"
                          name="Message"
                          placeholder="Add an optional message to the registration email sent to the user (max 2000 char)"
                          rows="6"
                        />
                        <div
                          class="message-counter"
                        />
                      </div>
                      <div
                        class="spacer"
                      />
                      <div
                        class="edit-icon"
                      >
                        <img
                          alt="Edit"
                          class="hide"
                          src="edit_24px.png"
                          style="width: auto; height: auto;"
                        />
                      </div>
                      <button
                        class="hide"
                        type="submit"
                      >
                        Save
                      </button>
                    </div>
                  </div>
                  <div
                    class="form-group d-flex justify-content-center"
                  >
                    <button
                      class="d-flex justify-content-center review-btn"
                      disabled=""
                      type="submit"
                    >
                      Review
                    </button>
                  </div>
                </form>
                <div
                  class="modal"
                  id="confirmModal"
                  role="dialog"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog modal-dialog-centered"
                  >
                    <div
                      class="modal-content"
                    >
                      <div
                        class="modal-body modal-open-text"
                      >
                        *Selected role for this member is
                         
                        <span
                          class="confirm-message"
                        />
                        , who will have
                         
                        <span
                          class="confirm-message"
                        />
                        <div
                          class="d-flex justify-content-center"
                        >
                          <button
                            class="modal-confirm-btn"
                            data-bs-dismiss="modal"
                            type="button"
                          >
                            Confirm
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
